packages/emf/backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/services/JSONResourceFactory.java

```diff
@@ -21,18 +21,43 @@
 import org.eclipse.sirius.emfjson.resource.JsonResourceImpl;
 
 /**
- * Special factory for {@link JsonResource} for Sirius Web.</br>
+ * Special factory for {@link JsonResource}.</br>
  * This factory ensures that objects in that resource are identified by their uuid.
  *
  * @author lfasani
  */
-public class SiriusWebJSONResourceFactoryImpl extends ResourceFactoryImpl {
+public class JSONResourceFactory extends ResourceFactoryImpl {
 
     @Override
     public JsonResource createResource(URI uri) {
         Map<String, Object> options = new HashMap<>();
```

Comment:
I propose we add this here:
```java
if (uri == null || uri.scheme() == null || uri.scheme().isBlank()) {
            throw new IllegalArgumentException(String.format("Missing scheme for URI %s", uri)); //$NON-NLS-1$
        }
```
to make any misuse fail early. Even if we review and fix all currently existing callers in client code (Sirius Web and others), it's pretty easy for this kind of misuse to reappear later on.

Reply:
done